system/include/emscripten/heap.h

```diff
@@ -31,10 +32,36 @@ uintptr_t *emscripten_get_sbrk_ptr(void);
 // cannot be used to shrink the size of the heap.
 int emscripten_resize_heap(size_t requested_size) EM_IMPORT(emscripten_resize_heap);
 
-// Returns the current size of the WebAssembly heap.
+// Returns the base address of the global data section. The global data section
+// contains the constant global and static function local data, either zero- or
+// non-zero initialized. This address stays constant throughout program lifetime.
+uintptr_t emscripten_get_global_base(void);
+
+// Returns the end address of the global data section. N.b. the memory location
+// pointed to by this end value is not part of the global data section itself, i.e.
+// the global data section consists of the half-open range [global_base,
+// global_end[ in the usual begin-end pointer span style. This address stays
+// constant throughout program lifetime.
+uintptr_t emscripten_get_global_end(void);
+
+// Returns the starting address of the dynamic memory heap. This value corresponds
+// to the the sbrk value at program startup, when no dynamic memory allocations
+// have yet been made. This address stays constant throughout program lifetime.
+uintptr_t emscripten_get_dyn_heap_base(void);
+
+// Returns the end address of the dynamic memory heap (same as the sbrk() point)
+// This value can change during program execution, as the WebAssembly heap is
+// grown by calling sbrk().
+#define emscripten_get_dyn_heap_end() (*emscripten_get_sbrk_ptr())
```

Comment:
Why implement his using a macro?

Reply:
The idea was to clearly convey that this "function" is a straightforward alias, and there is no need to have an extra function for this one. We can later at any point safely turn it into a function if needed.